yarn-project/circuit-types/src/tx/tx.ts

```diff
@@ -179,6 +179,15 @@ export class Tx extends Gossipable {
     return new TxHash(firstNullifier.toBuffer());
   }
 
+  /** Returns the tx hash, or undefined if none is set. */
+  tryGetTxHash(): TxHash | undefined {
+    try {
+      return this.getTxHash();
```

Comment:
Could this swallow a fundamental error?

Reply:
Thing is tx hash computation is going to change soon, so I didn't want to mess up with the `getTxHash` method